RedHatInsights/rhsm-subscriptions
# ENT-4845: Updates for billingAccountId in tally

https://issues.redhat.com/browse/ENT-4845

Most of the work was completed in ENT-4843 as part of this PR: https://github.com/RedHatInsights/rhsm-subscriptions/pull/981

**Test Steps:**

- Start app with `DEV_MODE=true SUBSCRIPTION_USE_STUB=true ./gradlew :bootRun`

Testing for Instance API:
- Run `./bin/insert-mock-hosts --num-aws 2 --billing-provider AWS --billing-account-id 456`
- Run
```
curl -X 'GET' \
  'http://localhost:8000/api/rhsm-subscriptions/v1/instances/products/RHEL?billing_account_id=123' \
  -H 'accept: application/json' \
  -H 'x-rh-identity: eyJpZGVudGl0eSI6eyJhY2NvdW50X251bWJlciI6ImFjY291bnQxMjMiLCJ0eXBlIjoiVXNlciIsInVzZXIiOnsiaXNfb3JnX2FkbWluIjp0cnVlfSwiaW50ZXJuYWwiOnsib3JnX2lkIjoib3JnMTIzIn19fQo='
```
- Should get billingAccountId in response data

Testing for changes to RhMarketplacePayloadMapper:

- Insert mock data:
```
INSERT INTO public.offering(
    sku, product_name, product_family, sla, usage, description, has_unlimited_usage)
    VALUES ('MW01882', 'OpenShift Streams for Apache Kafka', 'JBoss', 'Premium', 'Production', 'Red Hat OpenShift Streams for Apache Kafka (Hourly)', false);

INSERT INTO public.subscription(
	sku, owner_id, subscription_id, quantity, start_date, end_date, billing_provider_id, account_number, subscription_number, billing_provider, billing_account_id)
	VALUES ('MW01882', 'org123', '12351253', '1', '2012-05-13 16:32:00+00', '2032-05-13 16:32:00+00', 'testProductCode123;customer123', 'account123', '1235153', 'red hat', '456');
```
- Create tally-summary.json file with folliwing json:
`{"account_number":"account123","tally_snapshots":[{"id":"2b1ca6a5-2afe-4d84-9379-8a6d28a4b781","billing_provider":"red hat","billing_account_id":"456","snapshot_date":1652911085.530940000,"product_id":"rhosak","sla":"Premium","usage":"Production","granularity":"Hourly","tally_measurements":[{"uom":"Cores","value":1.0}]}]}`
- Run in kafka directory:`/bin/kafka-console-producer.sh --broker-list localhost:9092 --topic platform.rhsm-subscriptions.tally < tally-summary.json`
- Mine errored out with `Error submitting usage for snapshot IDs: 2b1ca6a5-2afe-4d84-9379-8a6d28a4b781
java.lang.IllegalArgumentException: RESTEASY003720: path param RH_MARKETPLACE_URL has not been provided by the parameter map` but confirmed in debugger that usage was created. If you get this error it means the usage was attempted to be sent.
